core/src/main/java/org/wso2/testgrid/core/command/RunTestPlanCommand.java

```diff
@@ -314,30 +322,36 @@ private String deriveLogFilePath(Product product, TestPlan testPlan) {
      * @return TestPlan object model
      */
     private TestPlan generateTestPlan(DeploymentPattern deploymentPattern, TestConfig testConfig, String testRepoDir,
-                                      String infraRepoDir) {
-        TestPlan testPlan = new TestPlan();
-        testPlan.setStatus(Status.PENDING);
-        testPlan.setInfraRepoDir(infraRepoDir);
-        testPlan.setTestRepoDir(testRepoDir);
-        testPlan.setDeploymentPattern(deploymentPattern);
-        testPlan.setInfraParameters(testConfig.getInfraParams().get(0).toString());
-        deploymentPattern.addTestPlan(testPlan);
-
-        // Set test run number
-        int latestTestRunNumber = getLatestTestRunNumber(deploymentPattern, testPlan.getInfraParameters());
-        testPlan.setTestRunNumber(latestTestRunNumber + 1);
-
-        // Set test scenarios
-        List<TestScenario> testScenarios = new ArrayList<>();
-        for (String name : testConfig.getScenarios()) {
-            TestScenario testScenario = new TestScenario();
-            testScenario.setName(name);
-            testScenario.setTestPlan(testPlan);
-            testScenario.setStatus(Status.PENDING);
-            testScenarios.add(testScenario);
+                                      String infraRepoDir) throws CommandExecutionException {
+        try {
+            String jsonInfraParams = new ObjectMapper().writeValueAsString(testConfig.getInfraParams().get(0));
+            TestPlan testPlan = new TestPlan();
+            testPlan.setStatus(Status.PENDING);
```

Comment:
We should take out the STATUS from TestPlan if possible. We can have the STATUS maintained in the same TestPlan table in the DB though.

Reply:
Fixed. Statuses will be set where appropriate.